src/main/scala/org/neo4j/spark/service/SchemaService.scala

```diff
@@ -158,6 +163,104 @@ class SchemaService(private val options: Neo4jOptions, private val jobId: String
     struct
   }
 
+  def countForNode(): Long = try {
+    /*
+     * we try to leverage the count store in order to have the faster response possible
+     * https://neo4j.com/developer/kb/fast-counts-using-the-count-store/
+     * so in this scenario we have some limitations given the fact that we get the min
+     * for the sequence of counts returned
+     */
+    val query = "CALL apoc.meta.stats() yield labels RETURN labels"
+    val map = session.run(query).single()
+      .asMap()
+      .asScala
+      .get("labels")
+      .getOrElse(Collections.emptyMap())
+      .asInstanceOf[util.Map[String, Long]].asScala
+    map.filterKeys(k => options.nodeMetadata.labels.contains(k))
+      .values.min
+  } catch {
+    case e: ClientException =>
+      e.code match {
+        case "Neo.ClientError.Procedure.ProcedureNotFound" => {
+          // TODO get back to Cypher DSL
+          options.nodeMetadata.labels
+            .map(label => {
+              val query = s"""MATCH (${Neo4jUtil.NODE_ALIAS}:$label)
```

Comment:
you can do a union query that uses 

`MATCH (:`$label`) RETURN { type: '$label', count:count(*) } as counts UNION ALL ...`